File: test/fixtures/small-php/Greeter.php
<?php

namespace Sample;

/** Greets people by name. */
class Greeter
{
    private string $name;

    public function __construct(string $name)
    {
        $this->name = $name;
    }

    /** Returns a greeting. */
    public function greet(): string
    {
        return $this->format();
    }

    private function format(): string
    {
        return normalize($this->name);
    }

    public function summary(): string
    {
        return $this->describe();
    }

    public function describe(): string
    {
        return $this->name;
    }
}

/** Normalizes a string (free function). */
function normalize(string $s): string
{
    return trim($s);
}

File: test/fixtures/small-cs/Greeter.cs
namespace Sample;

/// <summary>Greets people by name.</summary>
public class Greeter
{
    private readonly string _name;

    public Greeter(string name)
    {
        _name = name;
    }

    /// <summary>Returns a greeting.</summary>
    public string Greet()
    {
        return Format();
    }

    private string Format()
    {
        return Normalize(_name);
    }

    private string Normalize(string s)
    {
        return s.Trim();
    }

    public string Summary => Describe();

    public string Describe()
    {
        return _name;
    }
}
